presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -377,19 +398,113 @@ public TableStatistics getTableStatistics(ConnectorTransactionHandle transaction
+                            return OptionalDouble.of(columnStatistics.getDecimalColumnStatistics().getDistinctValuesCount());
+                        case BOOLEAN:
+                            return OptionalDouble.of((columnStatistics.getBooleanColumnStatistics().getTrueCount() > 0 ? 1 : 0) + (columnStatistics.getBooleanColumnStatistics().getFalseCount() > 0 ? 1 : 0));
+                        case DATE:
+                            return OptionalDouble.of(columnStatistics.getDateColumnStatistics().getDistinctValuesCount());
+                        case STRING:
+                            return OptionalDouble.of(columnStatistics.getStringColumnStatistics().getDistinctValuesCount());
+                        default:
+                            return OptionalDouble.empty();
+                    }
+                },
+                DoubleStream::max);
+    }
+
+    private Estimate calculateNullsCount(Map<String, PartitionStatistics> partitionStatisticsMap, String column)
+    {
+        return getColumnStatisticsValue(
+                partitionStatisticsMap,
+                column,
+                columnStatistics -> {
+                    switch (columnStatistics.getType()) {
+                        case LONG:
+                            return OptionalDouble.of(columnStatistics.getLongColumnStatistics().getNullsCount());
+                        case DOUBLE:
+                            return OptionalDouble.of(columnStatistics.getDoubleColumnStatistics().getNullsCount());
+                        case DECIMAL:
+                            return OptionalDouble.of(columnStatistics.getDecimalColumnStatistics().getNullsCount());
+                        case BOOLEAN:
+                            return OptionalDouble.of(columnStatistics.getBooleanColumnStatistics().getNullsCount());
+                        case DATE:
+                            return OptionalDouble.of(columnStatistics.getDateColumnStatistics().getNullsCount());
+                        case STRING:
+                            return OptionalDouble.of(columnStatistics.getStringColumnStatistics().getNullsCount());
+                        case BINARY:
+                            return OptionalDouble.of(columnStatistics.getBinaryColumnStatistics().getNullsCount());
+                        default:
+                            return OptionalDouble.empty();
+                    }
+                },
+                doubleStream -> {
+                    List<Double> values = doubleStream.mapToObj(Double::valueOf).collect(toList());
+                    long valuesCount = values.size();
+                    double valuesSum = values.stream().mapToDouble(Double::doubleValue).sum();
+                    if (valuesCount == 0) {
+                        return OptionalDouble.empty();
+                    }
+                    else {
+                        return OptionalDouble.of(partitionStatisticsMap.size() / valuesCount * valuesSum);
```

Comment:
is this all (with and w/o stats) partition count: `partitionStatisticsMap.size()`? Make it more clear